scripts/generate_samples.ts

```diff
@@ -0,0 +1,88 @@
+import fs from "node:fs";
+import { chromium } from "@playwright/test";
+
+// Declare a global augmentation for the Window interface
+declare global {
+  interface Window {
+    map?: {
+      loaded: () => boolean;
+    };
+  }
+}
+
+type LocationClip = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+type SampleSpecification = {
+  location: string;
+  name: string;
+  viewport: {
+    width: number;
+    height: number;
+  };
+  clip: LocationClip;
+};
+
+// Load list of locations to take map screenshots
+const loadSampleLocations = (filePath: string): SampleSpecification[] => {
+  const rawData = fs.readFileSync(filePath, "utf8");
+  return JSON.parse(rawData);
+};
+
+const sampleFolder = "./samples";
+
+const jsonSampleLocations = process.argv[2] ?? "test/sample_locations.json";
+
+console.log(`Loading sample locations from ${jsonSampleLocations}`);
+
+const screenshots: SampleSpecification[] =
+  loadSampleLocations(jsonSampleLocations);
+
+fs.mkdirSync(sampleFolder, { recursive: true });
+
+const browser = await chromium.launch({ headless: true });
```

Comment:
```suggestion
const browser = await chromium.launch({
  headless: true,
  executablePath: process.env.CHROME_BIN,
});
```

Reply:
Will this work for both documented methods of `npx playwright install chromium` and `export CHROME_BIN=/usr/bin/chromium` ?

Reply:
Yes, I tested it; we could omit the variable if it's undefined but that seems unnecessary.